### PR TITLE
Fixes #1382. Update NRL solar table read code

### LINKED ISSUE #1382: Is there a way to avoid these broadcast calls?
https://github.com/GEOS-ESM/MAPL/blob/0dbe0f14adf059e53d29112d0ee2da9f4d560275/base/MAPL_sun_uc.F90#L2748-L2753

Can we do some kind of parallel read here instead of having the root node broadcast everything.  This is trapping a load imbalance from the moist physics leading to a significant overhead in Surface times.

Every PE is opening and closing the file currently as it is written

### PR DESCRIPTION
**Pull request by mathomp4**
## Description

This PR updates the `MAPL_SunGetSolarConstantFromNRLFile` procedure to:

1. Open and read the file only on root
2. Broadcast the *tables* instead of just the three output values
3. Does interpolation on all processes to avoid broadcast at every step

## Related Issue

Closes #1382 

## Motivation and Context

Reduces some blocking MPI calls on every run step.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [x] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Comment by mathomp4** [comment]
As far as my testing can see, this is zero-diff. I'll undraft and let @dr0cloud and @wmputman take a look.

File: base/MAPL_sun_uc.F90 [hunk 1]
```diff
@@ -2475,54 +2475,45 @@ subroutine MAPL_SunGetSolarConstantFromNRLFile(CLOCK,filename_in,SC,MG,SB,Persis
...
+         ! number of lines so we can allocate our tables
+         ! ---------------------------------------------
 
          if (amIRoot) then
 
+            ! Open the file
+            ! -------------
+            filename = trim(filename_in)
+            open(newunit=unit, file=filename, form="formatted", status="old", iostat=status)
+            _ASSERT(status==0,'Could not find NRL data file '// filename )
```
Comment (tclune) [inline base/MAPL_sun_uc.F90:2488]: Need a `trim()` on filename here.
```suggestion
            _ASSERT(status==0,'Could not find NRL data file '// trim(filename ))
```

**Review by tclune — APPROVED** [review]

**Comment by dr0cloud** [comment]
@mathomp4 Sorry, I left early Friday so didn't get a chance to review. I have reviewed your changes unofficially and they seem fine. I don't think I can review in git anymore, right? Re the other two issues I mentioned, "belt and suspenders check" and INDX1&2 ASSERTs, let's discuss on Teams chat Tuesday.

**Comment by mathomp4** [comment]
> @mathomp4 Sorry, I left early Friday so didn't get a chance to review. I have reviewed your changes unofficially and they seem fine. I don't think I can review in git anymore, right? Re the other two issues I mentioned, "belt and suspenders check" and INDX1&2 ASSERTs, let's discuss on Teams chat Tuesday.

I've pushed a reloop of the code, @dr0cloud. Please take a look.

**Comment by mathomp4** [comment]
@dr0cloud The two asserts are now in as well. Seems to work.

**Review by tclune — APPROVED** [review]

**Pull request merged**